@kiva/kv-shop/src/trackTransaction.ts

```diff
@@ -0,0 +1,75 @@
+import { gql, type ApolloClient } from '@apollo/client/core';
+import { callShopMutation } from './shopQueries';
+import { getVisitorID } from './util/visitorId';
+
+import parseSPCookie from './util/parseSPCookie';
+
+export const trackTransactionMutation = gql`
+	mutation trackTransaction(
+		$campaign: String,
+		$campaignContent: String,
+		$medium: String,
+		$snowplowUserId: String,
+		$snowplowSessionId: String,
+		$source: String,
+		$transactionId: Int!,
+		$visitorId: String
+	) {
+		shop {
+			id
+			trackTransaction(
+				campaign: $campaign
+				campaignContent: $campaignContent
+				medium: $medium
+				snowplowUserId: $snowplowUserId
+				snowplowSessionId: $snowplowSessionId
+				source: $source
+				transactionId: $transactionId
+				visitorId: $visitorId
+			)
+		}
+	}
+`;
+
+export interface TrackTransactionData {
+	shop: {
+		id: string,
+		trackTransaction: boolean,
+	} | null,
+}
+
+export interface TrackTransactionOptions {
+	apollo: ApolloClient<any>,
+	transactionId: string,
+}
+
+export async function trackTransactionEvent({
+	apollo,
+	transactionId,
+}: TrackTransactionOptions) {
+	if (!transactionId) {
+		// exit if missing transaction id
+		return false;
+	}
+
+	// NOTE: GA tracking removed due to __utmz cookie deprecation
```

Comment:
Rather than bring over the existing ga cookie parser we have in UI, I've just noted that it's deprecated as the `__utmz` cookie is no longer being generated or populated in our pages. Verified with data eng, that we're ok to pass in null for those values as there is no current way to retrieve them.